src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -506,12 +523,13 @@ class DrawSketchHandlerLine: public DrawSketchHandler
 
     virtual bool pressButton(Base::Vector2d onSketchPos) override
     {
+        toolWidgetManager.overrideSketchPosition(onSketchPos);
+
         if (Mode==STATUS_SEEK_First){
             EditCurve[0] = onSketchPos;
 
-            Mode = STATUS_SEEK_Second;
-
-            toolWidgetManager.updateParameterValues(); // let the widget know updated values
+            Mode = STATUS_SEEK_Second; 
+            toolWidgetManager.updateFocus();

```

Comment:
I have a small conflict with the name of this function.

Ideally (not always possible), the name should reflect what is asked on the object on which it is called. The name should not reflect the actual implementation (updating the focus of yet another object), rather how it affects the object.

Then, looking at the implementation of the function, it is actually not able to adapt to the focus in any situation. For example, we may be "seeking" second, but with the user having already set parameter number 3 via the widget. The thing is that there is an expectation from the client (DSH), that when this function is called, the focus is set in the right parameter. However, this function is unable to guarantee this outcome.

I would not be such a pain in the back, if it weren't because these functions are going to define an interface for all other handlers, therefore the importance of conveying the right intent...

Do not change anything at this time.

Reply:
The goal of this function is the following. Without it the following happens : 
If you select the first point manually by clicking somewhere. Then the third spinbox doesn't get the focus (the focus is even lost from the first spinbox because of the click). So you can't type the coordinate of second point directly if you want to.
The desired behaviour, is that if the user doesn't use ToolSettings for one step (for status_seek_first for example), then ToolSettings should get ready to accept parameters for status_seek_second when the mode changes.